packages/js/product-editor/src/hooks/use-product-attributes.ts

```diff
@@ -83,7 +83,7 @@ export function useProductAttributes( {
 		variation: boolean,
 		startPosition: number
 	): ProductAttribute[] => {
-		return atts.map( ( { isDefault, terms, ...attribute }, index ) => ( {
+		return atts.map( ( { terms, ...attribute }, index ) => ( {
```

Comment:
This is what fixed the `Set default value` issue, as it was removed from the attribute here, always equating to `false`.